src/spots/spots.controller.ts

```diff
@@ -144,19 +115,19 @@ export class SpotsController {
 		responseType: ReadSpotDto,
 		operationId: 'updateSpot',
 	})
-	@TokenOrJwtAuth()
+	@UnconfiguredSpotOrTokenOrJwtAuth()
 	async updateSpot(
 		@AuthUser() user: User,
 		@Param('id') id: string,
 		@Body() updateSpot: UpdateSpotDto,
 	): Promise<ReadSpotDto> {
-		if (updateSpot.referenced) {
+		if (updateSpot.public) {
 			const isCertifiedOrAdmin =
 				user.role === UserRole.CERTIFIED_USER || user.role === UserRole.ADMIN;
 
 			if (!isCertifiedOrAdmin) {
 				throw new ForbiddenException(
-					'Standard users cannot create referenced spots',
+					'Standard users cannot create public spots',
 				);
```

Comment:
If a user creates an unconfigured private spot and wants to update it to pass it public, the system will not allow it as it is considerated an unconfigured spot, thus the role of the user is changed to `GUEST` (as per the `UnconfiguredSpotOrTokenOrJwtAuth` guard), disallowing the user to update its unconfigured spot from private to public.